modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -188,8 +191,88 @@ PolygonMeshGeneratorBase::buildSlice(
+               "PolygonMeshGeneratorBase::buildSlice().");
+  // In order to create quadratic elements (i.e., order = 2), we creates nodes with double mesh
+  // density. Thus, the related parameters need to be modified accordingly. A prefix "mod_" is used
+  // to indicate the modified parameters.
+
+  // For ring_layers, modification is to double the number of layers for order = 2
+  std::vector<unsigned int> mod_ring_layers(ring_layers);
+  std::for_each(
+      mod_ring_layers.begin(), mod_ring_layers.end(), [&order](unsigned int & n) { n *= order; });
+  // For ring_radial_biases, modification is to take the square root of the original biases for
+  // order = 2
+  std::vector<Real> mod_ring_radial_biases(ring_radial_biases);
+  std::for_each(mod_ring_radial_biases.begin(),
+                mod_ring_radial_biases.end(),
+                [&order](Real & n) { n = std::pow(n, 1.0 / order); });
+  // ducts_layers is similar to ring_layers
+  std::vector<unsigned int> mod_ducts_layers(ducts_layers);
+  std::for_each(
+      mod_ducts_layers.begin(), mod_ducts_layers.end(), [&order](unsigned int & n) { n *= order; });
+  // duct_radial_biases is similar to ring_radial_biases
+  std::vector<Real> mod_duct_radial_biases(duct_radial_biases);
+  std::for_each(mod_duct_radial_biases.begin(),
+                mod_duct_radial_biases.end(),
+                [&order](Real & n) { n = std::pow(n, 1.0 / order); });
+  // Azimuthal mesh density is also doubled for order = 2
+  const unsigned int mod_num_sectors_per_side = num_sectors_per_side * order;
+  const unsigned int mod_background_intervals = background_intervals * order;
+  // background_radial_bias is similar to ring_radial_biases
+  const Real mod_background_radial_bias = std::pow(background_radial_bias, 1.0 / order);
+  // Perform similar modifications for boundary layer parameters
+  multiBdryLayerParams mod_ring_inner_boundary_layer_params(ring_inner_boundary_layer_params);
+  std::for_each(mod_ring_inner_boundary_layer_params.intervals.begin(),
+                mod_ring_inner_boundary_layer_params.intervals.end(),
+                [&order](unsigned int & n) { n *= order; });
+  std::for_each(mod_ring_inner_boundary_layer_params.biases.begin(),
+                mod_ring_inner_boundary_layer_params.biases.end(),
+                [&order](Real & n) { n = std::pow(n, 1.0 / order); });
+  multiBdryLayerParams mod_ring_outer_boundary_layer_params(ring_outer_boundary_layer_params);
+  std::for_each(mod_ring_outer_boundary_layer_params.intervals.begin(),
+                mod_ring_outer_boundary_layer_params.intervals.end(),
+                [&order](unsigned int & n) { n *= order; });
+  std::for_each(mod_ring_outer_boundary_layer_params.biases.begin(),
+                mod_ring_outer_boundary_layer_params.biases.end(),
+                [&order](Real & n) { n = std::pow(n, 1.0 / order); });
+  multiBdryLayerParams mod_duct_inner_boundary_layer_params(duct_inner_boundary_layer_params);
```

Comment:
can we have a routine defined on multiBdryLayerParams and singleBdryLayerParams that performs all these operations?